File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/test/model/annotation/AbstractObjectTest.java
package com.digirati.elucidate.common.test.model.annotation;

import com.digirati.elucidate.common.model.annotation.AbstractObject;
import com.digirati.elucidate.common.test.AbstractTest;

import java.util.Date;
import java.util.Map;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertNotNull;

public abstract class AbstractObjectTest<T extends AbstractObject> extends AbstractTest {

    protected void testAbstractObject(T abstractObject) {

        assertNotNull(abstractObject);

        Date createdDateTime = generateRandomDate();
        abstractObject.setCreatedDateTime(createdDateTime);
        assertThat(createdDateTime, is(equalTo(abstractObject.getCreatedDateTime())));

        boolean deleted = generateRandomBoolean();
        abstractObject.setDeleted(deleted);
        assertThat(deleted, is(equalTo(abstractObject.isDeleted())));

        Map<String, Object> jsonMap = generateRandomJsonMap();
        abstractObject.setJsonMap(jsonMap);
        assertThat(jsonMap, is(equalTo(abstractObject.getJsonMap())));

        Date modifiedDateTime = generateRandomDate();
        abstractObject.setModifiedDateTime(modifiedDateTime);
        assertThat(modifiedDateTime, is(equalTo(abstractObject.getModifiedDateTime())));
    }
}

File: elucidate-common-lib/src/test/java/com/digirati/elucidate/common/test/model/annotation/AbstractObjectTest.kt
package com.digirati.elucidate.common.test.model.annotation

import com.digirati.elucidate.common.model.annotation.AbstractObject
import com.digirati.elucidate.common.test.AbstractTest
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers
import org.hamcrest.Matchers.equalTo
import org.junit.Assert.assertNotNull

abstract class AbstractObjectTest<T : AbstractObject?> : AbstractTest() {
    protected fun testAbstractObject(abstractObject: T) {
        assertNotNull(abstractObject)
        val createdDateTime = generateRandomDate()
        abstractObject!!.createdDateTime = createdDateTime
        assertThat(
            createdDateTime, Matchers.`is`(
                equalTo(
                    abstractObject.createdDateTime
                )
            )
        )
        val deleted = generateRandomBoolean()
        abstractObject.isDeleted = deleted
        assertThat(
            deleted, Matchers.`is`(
                equalTo(
                    abstractObject.isDeleted
                )
            )
        )
        val jsonMap = generateRandomJsonMap()
        abstractObject.jsonMap = jsonMap
        assertThat(
            jsonMap, Matchers.`is`(
                equalTo(
                    abstractObject.jsonMap
                )
            )
        )
        val modifiedDateTime = generateRandomDate()
        abstractObject.modifiedDateTime = modifiedDateTime
        assertThat(
            modifiedDateTime, Matchers.`is`(
                equalTo(
                    abstractObject.modifiedDateTime
                )
            )
        )
    }
}